addons/account/models/company.py

```diff
@@ -651,48 +654,58 @@ def _check_hash_integrity(self):
                 'sequence_number',
                 'secure_sequence_number',
             ]
-            moves = self.env['account.move'].sudo().search_fetch(
-                domain=[
-                    ('journal_id', '=', journal.id),
-                    ('inalterable_hash', '!=', False),
-                ],
-                field_names=field_names,
-                order="secure_sequence_number ASC NULLS LAST, sequence_prefix, sequence_number ASC"
-            )
-
-            if not moves:
-                results.append({
-                    'journal_name': journal.name,
-                    'restricted_by_hash_table': 'V',
-                    'status': 'no_data',
-                    'msg_cover': _('There is no journal entry flagged for accounting data inalterability yet.'),
-                })
-                continue
 
+            offset = 0
+            stop = False
             prefix2result = defaultdict(lambda: {
                 'first_move': self.env['account.move'],
                 'last_move': self.env['account.move'],
                 'corrupted_move': self.env['account.move'],
             })
             last_move = self.env['account.move']
-            current_hash_version = 1
-            for move in moves:
-                prefix_result = prefix2result[move.sequence_prefix]
-                if prefix_result['corrupted_move']:
+            while not stop:
+                moves = self.env['account.move'].sudo().search_fetch(
```

Comment:
Using `search_fetch` will still fill the ORM's cache without clearing it. It will also need to redo the query multiple times because of the offset mechanism.
Using `_search` with `dictfetchmany` can solve the issue for real by allowing the garbage collector to do its job and by using a serverside cursor.

Reply:
Done, I had to use `fetchmany` (the first occurence in the database ever wow) instead and then browse the records because later on in the code we need to have objects as dict of values is not sufficient

https://github.com/odoo/odoo/blob/d32ba694afe6b9ce365d5b93e587dffc852afd27/addons/account/models/account_move.py#L3315-L3318